packages/js/components/src/form/form.tsx

```diff
@@ -220,9 +207,7 @@ function FormComponent< Values extends Record< string, any > >(
 	const setValue = useCallback(
 		// eslint-disable-next-line @typescript-eslint/no-explicit-any
 		( name: string, value: any ) => {
-			const valuesToSet: Values = {} as Values;
-			valuesToSet[ name as keyof Values ] = value;
-			setValues( valuesToSet );
+			setValues( _setWith( { ...values }, name, value, _clone ) );
```

Comment:
cc @joshuatf  in relation to the `lodash` usage, I know you are less of a fan of using lodash.
This does allow us to reference fields in this manner `dimensions.width` which is very handy, for the shipping field for example.
We could potentially implement this manually, but given lodash would be much more stable, I would be fine with using this (given we aren't pruning it out yet).
But let me know if you have any strong sense against this?